### PR TITLE
Add one master CI which builds, tests and deploys

### PR DESCRIPTION
**Pull request by paddyroddy**
This is a solution to solving #192, however we have not used `cibuildwheels`. Some issues on that to follow (#235), and I'll explain why here.

Using `cibuildwheels` to do everything explained in #192 proved quite complex and would involve a lot of changes. This PR has the minimal changes which respect the current workflow but remove the libraries from `git`. I will raise an issue separately about Windows shortly (#234). We (@p-j-smith and I) hadn't realised that the Windows binary was cross-compiled, so a bit of pain trying to fix that.

The main thing done here is that now where we previously had a testing workflow and a deployment workflow, we now build the C++, test the python and deploy in a single workflow. This should ensure we don't have a failing deployment.

### CODE REVIEW

**Comment by paddyroddy** [comment]
@quantumjot can you test the deployment? Perhaps when #228 is merged, seeing as you are bumping the version?

File: .github/workflows/build_wheels_test_and_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            path: |
+              btrack/libs/*
+            if-no-files-found: error
+
+  test:
+    needs: build_wheels
+    name: ${{ matrix.os }} py${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
```
Comment (p-j-smith) [inline .github/workflows/build_wheels_test_and_deploy.yml:53]: maybe set this to `true` now that everything is working

File: .github/workflows/build_wheels_test_and_deploy.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+          cache-dependency-path: "pyproject.toml"
+
+      - name: Install build dependencies
+        run: python -m pip install build wheel
+
+      - name: Build distributions
+        shell: bash -l {0}
+        run: python -m build
+
+      - name: Publish package to PyPI
```
Comment (p-j-smith) [inline .github/workflows/build_wheels_test_and_deploy.yml:132]: we could also upload to Test PyPI when merging each PR: https://github.com/marketplace/actions/pypi-publish#advanced-release-management but that's probably best for a separate pr?

**Review by p-j-smith — CHANGES_REQUESTED** [review]
great! it's much better having this run in ci, and a good first step toward using cibuildwheel. One minor change about `fail-fast` in ci

File: .github/workflows/build_wheels_test_and_deploy.yml (hunk 2 above)
Comment (paddyroddy) [reply .github/workflows/build_wheels_test_and_deploy.yml:131]: quantumjot/btrack#237

File: Makefile [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
 ifeq ($(UNAME), Linux)
-	# do something Linux #-fopenmp -static
-	CXX = g++
-	EXT = so
-	XLDFLAGS = -Wl,--no-undefined -Wl,--no-allow-shlib-undefined
-	#-L/usr/local/cuda/lib64 -lcuda -lcudart
+	ifeq ($(WINDOWS_CROSS_COMPILE),true)
+		# do something Windowsy
+		CXX = x86_64-w64-mingw32-g++
+		EXT	 = DLL
```
Comment (quantumjot) [inline Makefile:7]:
```suggestion
		EXT = DLL
```

File: Makefile [hunk 4]
```diff
@@ -29,10 +30,10 @@ VERSION_MAJOR = $(shell cat $(VERSION_FILE) | cut -f1 -d.)
...
 
-# If your compiler is a bit older you may need to change -std=c++11 to -std=c++0x
+# If your compiler is a bit older you may need to change -std=c++17 to -std=c++0x
 #-I/usr/include/python2.7 -L/usr/lib/python2.7 # -O3
 LLDBFLAGS =
-CXXFLAGS = -c -std=c++11 -m64 -fPIC -I"./btrack/include" \
+CXXFLAGS = -c -std=c++17 -m64 -fPIC -I"./btrack/include" \
 					 -DDEBUG=false -DVERSION_MAJOR=$(VERSION_MAJOR) \
 					 -DVERSION_MINOR=$(VERSION_MINOR) -DVERSION_BUILD=$(VERSION_BUILD) \
 					 -DBUILD_SHARED_LIB
```
Comment (quantumjot) [inline Makefile:39]: Looks like the formatting is off here

**Review by quantumjot — APPROVED** [review]
Great stuff - a big improvement already. Once this is merged, I can take a look with #228 .

File: Makefile (hunk 4 above)
Comment (paddyroddy) [reply Makefile]: It's the same as before but have changed

**Pull request merged**